controlplane/kubeadm/internal/workload_cluster_etcd.go

```diff
@@ -45,7 +45,7 @@ func (w *Workload) ReconcileEtcdMembers(ctx context.Context, nodeNames []string,
 		if err != nil {
 			continue
 		}
-		defer etcdClient.Close()
+		defer etcdClient.Close() //nolint:gocritic // etcdClient is closed correctly, no resource leak.
```

Comment:
Assuming gocritic doesn't like defers in a for loop, which we should probably avoid. Can we use an inline function or move this code somewhere else so the defer is called right away when the function ends?

Reply:
Yup the issue is that it doesn't like defer close in for loops. Moved to a new func.